tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -2625,5 +2723,177 @@ Status StorageManagerCanonical::group_metadata_vacuum(
   return consolidator->vacuum(group_name);
 }
 
+void StorageManager::encryption_key_from_configs(EncryptionKey &enc_key, const Config* config) {
```

Comment:
This is a good idea and there are a lot of places in storage manager that do exactly this. Could we replace them as well. Also, any reason this code cannot live in Config to alleviate the already overloaded StorageManager class?

Reply:
Currently changed per  EricHughes suggestion as EncryptionKey constructor

Reply:
Not changing everywhere at this time as it would entangle this entire PR if one of those areas should be changed incorrectly.

(story 23490 created)